### PR TITLE
docs: add example `composer.json` for required cache adapters

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| Documentation | yes

### Description

With this example, it is possible to only require those adapters which are required by the project.

### CODE REVIEW

**Comment by froschdesign** [comment]
@boesing 
_Sorry for the late response!_

These are migration notes, right? I would suggest to create a page in the documentation and add this information there. Additionally, a link to the migration page should be added in the readme file.

**Comment by boesing** [comment]
@froschdesign No, this is just an information on how to configure a current 2.10 based project to not install all unnecessary cache adapters pre-3.0.
I dont think we can provide PHP 8 support to all adapters as most of them are already not supported on PHP 7 either.

So, we dont have to migrate all adapters to PHP 8 at once and let people decide on which adapters they use and to provide PRs only for these adapters.

So in the end, it is forward compatibility to 3.0 aswell as PHP 8 not having to be supported on every adapter for 2.10+.
Thats the idea behind this example.

**Comment by froschdesign** [comment]
> No, this is just an information on how to configure a current 2.10 based project to not install all unnecessary cache adapters pre-3.0.

Then the description with the example should be added to the installation page.

**Comment by boesing** [comment]
@froschdesign Just to be sure I understand this correct, we dont want to provide informations in `README.md` but via docs only? 

I've also seen that this package is not using the `installation` template other packages are using. So yes, I can extend the installation page here, but how to achieve that in packages which are using the installation template?
Shouldn't we use the template in each package?

**Comment by froschdesign** [comment]
@boesing 
> Just to be sure I understand this correct, we dont want to provide informations in `README.md` but via docs only?

Oh no. In the optimum case we should provide the same informations (like intro, installation, etc.) in the readme file (here on GitHub and after the installation in the vendor directory) and in the documentation, without duplicating the content. Because duplicating always leads to forgotten updates on one side.
So far, we don't have a (good) solution for this, so it reduces to some text and links.

> Shouldn't we use the template in each package?

Yes. I would provide a solution that allows custom content to be added, like the Composer configuration.

File: README.md [hunk 1]
```diff
@@ -11,6 +11,66 @@ storage adapters (DB, File, Memcache, etc).
 - File issues at https://github.com/laminas/laminas-cache/issues
 - Documentation is at https://docs.laminas.dev/laminas-cache/
 
+## Avoid unused cache adapters are being installed
+With `laminas-cache` v2.10.0, we introduced satellite packages for all cache adapters.
```
Comment (froschdesign) [inline README.md]:
```suggestion
## Avoid Unused Cache Adapters Are Being Installed

With `laminas-cache` v2.10.0, we introduced satellite packages for all cache adapters.
```

File: README.md [hunk 2]
```diff
@@ -11,6 +11,66 @@ storage adapters (DB, File, Memcache, etc).
...
+
+In case, there is no need for several adapters in your project, you can use composer to ensure these adapters are not being installed. To make this happen, you have to specify a `replace` property within the `composer.json` of your project.
+
+### Example composer.json with only memory adapter being installed
+```json
```
Comment (froschdesign) [inline README.md]:
```suggestion
### Example `composer.json` with Only Memory Adapter Being Installed

```json
```

File: README.md [hunk 3]
```diff
@@ -11,6 +11,66 @@ storage adapters (DB, File, Memcache, etc).
...
+        "laminas/laminas-cache-storage-adapter-session": "*",
+        "laminas/laminas-cache-storage-adapter-wincache": "*",
+        "laminas/laminas-cache-storage-adapter-xcache": "*",
+        "laminas/laminas-cache-storage-adapter-zend-server": "*"
+    }
+}
+```
+
+> ### Only necessary in 2.10+
+> Starting with 3.0.0, no storage adapter is required by this component and thus, each project has to specify the storage adapters which are required by the project.
```
Comment (froschdesign) [inline README.md]:
```suggestion
> ### Only necessary in 2.10+
>
> Starting with 3.0.0, no storage adapter is required by this component and thus, each project has to specify the storage adapters which are required by the project.
```

File: docs/book/installation.md [hunk 4]
```diff
@@ -1,5 +1,63 @@
-**This is only a placeholder.**
+---
+show_file_content: true
+---
 
-The content of this page can be found under:
+## Avoid unused cache adapters are being installed
+With `laminas-cache` v2.10.0, we introduced satellite packages for all cache adapters.
```
Comment (froschdesign) [inline docs/book/installation.md]:
```suggestion
## Avoid Unused Cache Adapters Are Being Installed

With `laminas-cache` v2.10.0, we introduced satellite packages for all cache adapters.
```

File: docs/book/installation.md [hunk 5]
```diff
@@ -1,5 +1,63 @@
...
+        "laminas/laminas-cache-storage-adapter-session": "*",
+        "laminas/laminas-cache-storage-adapter-wincache": "*",
+        "laminas/laminas-cache-storage-adapter-xcache": "*",
+        "laminas/laminas-cache-storage-adapter-zend-server": "*"
+    }
+}
+```
+
+> ### Only necessary in 2.10+
+> Starting with 3.0.0, no storage adapter is required by this component and thus, each project has to specify the storage adapters which are required by the project.
```
Comment (froschdesign) [inline docs/book/installation.md]:
```suggestion
> ### Only necessary in 2.10+
>
> Starting with 3.0.0, no storage adapter is required by this component and thus, each project has to specify the storage adapters which are required by the project.
```

File: docs/book/installation.md [hunk 6]
```diff
@@ -1,5 +1,63 @@
...
-The content of this page can be found under:
+## Avoid unused cache adapters are being installed
+With `laminas-cache` v2.10.0, we introduced satellite packages for all cache adapters.
 
-https://github.com/laminas/documentation-theme/blob/master/theme/pages/installation.html
\ No newline at end of file
+In case, there is no need for several adapters in your project, you can use composer to ensure these adapters are not being installed. To make this happen, you have to specify a `replace` property within the `composer.json` of your project.
+
+### Example composer.json with only memory adapter being installed
+```json
```
Comment (froschdesign) [inline docs/book/installation.md]:
```suggestion
### Example `composer.json` with Only Memory Adapter Being Installed

```json
```

File: docs/book/installation.md (hunk 5 above)
Comment (froschdesign) [reply docs/book/installation.md]: Move this block quote to the beginning as it breaks the example and all version 3 users can skip this part / information from the beginning.
(after the headline "Avoid Unused Cache Adapters Are Being Installed")

File: README.md (hunk 3 above)
Comment (froschdesign) [reply README.md]: Move this block quote to the beginning as it breaks the example and all version 3 users can skip this part / information from the beginning.
(after the headline "Avoid Unused Cache Adapters Are Being Installed")

**Review by froschdesign — CHANGES_REQUESTED** [review]
Some small changes are needed:

* For headline we uses the "[Chicago Manual of Style Capitalization Rules](https://capitalizemytitle.com/#ChicagoStyle)"
* [MD022 - Headings should be surrounded by blank lines](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md022)

**Comment by boesing** [comment]
Bummer! I sweared to myself that I will keep an eye on these chicago headlines 😂

**Review by froschdesign — APPROVED** [review]
🚢

**Pull request merged**